dll/win32/shell32/CDefView.cpp

```diff
@@ -3039,11 +3074,9 @@ HRESULT WINAPI CDefView::Drop(IDataObject* pDataObject, DWORD grfKeyState, POINT
     ImageList_DragLeave(m_hWnd);
     ImageList_EndDrag();
 
-    if ((m_iDragOverItem == -1 || m_pCurDropTarget == NULL) && 
+    if ((IsDropOnSource(NULL) == S_OK) && 
         (*pdwEffect & DROPEFFECT_MOVE) && 
-        /*(GetKeyState(VK_LBUTTON) != 0) &&*/
-        (m_pSourceDataObject.p) && 
-        (SHIsSameObject(pDataObject, m_pSourceDataObject)))
+        (m_grfKeyState & MK_LBUTTON))
```

Comment:
OK so there's no need for SHIsSameObject call anymore?

Reply:
No, I don't think it was needed in the first place. m_pSourceDataObject is set by the drag source and CDefView::Drop is called at the drop target. This means that in a drag operation there can be two CDefView involved (but it is not nescessary that the source is always a CDefView). The target knows that it is a drag operation in the same control just by seeing that m_pSourceDataObject has a value. This means that a drag operation is active from the target. That's why I think the check for the data object is superfluous.